plugins/woocommerce/includes/data-stores/class-wc-order-data-store-cpt.php

```diff
@@ -184,8 +185,17 @@ protected function read_order_data( &$order, $post_object ) {
 	 * @param WC_Order $order Order object.
 	 */
 	public function update( &$order ) {
+		/**
+		 * Filter the order status to use when payment is complete.
+		 *
+		 * @since 3.0.0
+		 *
+		 * @param string   $payment_complete_status Default status to use when payment is complete.
+		 * @param int      $order_id               Order ID.
+		 */
+		$payment_complete_status = apply_filters( 'woocommerce_payment_complete_order_status', $order->needs_processing() ? OrderStatus::PROCESSING : OrderStatus::COMPLETED, $order->get_id(), $order );
```

Comment:
Filter invocation frequency change.

Reply:
Fixed in https://github.com/woocommerce/woocommerce/pull/51922/commits/9071d081dfe753615780eec18284c19023d255c9